modules/462-loki/templates/statefulset.yaml

```diff
@@ -66,29 +86,97 @@ spec:
             mountPath: /loki
           - name: tmp
             mountPath: /tmp
-        ports:
-          - name: http-metrics
-            containerPort: 3100
-            protocol: TCP
         livenessProbe:
           httpGet:
             path: /ready
-            port: http-metrics
+            port: 3100
+            scheme: HTTPS
           initialDelaySeconds: 45
         readinessProbe:
           httpGet:
             path: /ready
-            port: http-metrics
+            port: 3100
+            scheme: HTTPS
           initialDelaySeconds: 45
         resources:
           {{ include "helm_lib_resources_management_pod_resources" (list $.Values.loki.resourcesManagement) | nindent 10 }}
+      - name: kube-rbac-proxy
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all" . | nindent 8 }}
+        image: {{ include "helm_lib_module_common_image" (list . "kubeRbacProxy") }}
+        args:
+          - "--secure-listen-address=$(KUBE_RBAC_PROXY_LISTEN_ADDRESS):3100"
+          - "--client-ca-file=/etc/kube-rbac-proxy/ca.crt"
+          - "--v=2"
+          - "--logtostderr=true"
+          - "--stale-cache-interval=1h30m"
+        env:
+          - name: KUBE_RBAC_PROXY_LISTEN_ADDRESS
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          - name: KUBE_RBAC_PROXY_CONFIG
+            value: |
+              excludePaths:
+              - /ready
+              upstreams:
+              - upstream: http://127.0.0.1:3101/ready
+                path: /ready
+              - upstream: http://127.0.0.1:3101/metrics
+                path: /metrics
+                authorization:
+                  resourceAttributes:
+                    namespace: d8-monitoring
+                    apiGroup: apps
+                    apiVersion: v1
+                    resource: statefulsets
+                    subresource: prometheus-metrics
+                    name: loki
+              - upstream: http://127.0.0.1:3101/loki/api/v1/push
+                path: /loki/api/v1/push
+                authorization:
+                  resourceAttributes:
+                    namespace: d8-monitoring
+                    apiGroup: apps
+                    apiVersion: v1
+                    resource: statefulsets
+                    subresource: log-shipper
+                    name: loki
+              {{- include "grafana_loki_upstream" (list "/loki/api/v1/query" "/loki/api/v1/query_range" "/loki/api/v1/labels" "/loki/api/v1/label" "/loki/api/v1/label/" "/loki/api/v1/series" "/loki/api/v1/index/stats" "/loki/api/v1/index/volume" "/loki/api/v1/index/volume_range" "/loki/api/v1/tail" "/api/prom/tail" "/api/prom/query" "/api/prom/label" "/api/prom/label/" "/api/prom/series") | nindent 14 }}
```

Comment:
Why we use list of uri ? Can we give ro access to grafana at all ? For example, limit to  only GET requests ?

Reply:
- Limiting access to GET requests only is unsafe.
- `kube-rbac-proxy` does not support filtering using HTTP methods.